### PR TITLE
refactor: fixes gocritic complains

### LINKED ISSUE #543: lint: fix gocritic
* Fix `gocritic` warnings from the [lint checker](https://github.com/ovrclk/akash/blob/master/.github/workflows/tests.yml#L39) step for it.
 * Remove `continue-on-error: true` from `lint gocritic` step.

### PR DESCRIPTION
**Pull request by troian**
fixes #543

combine if statements executing same bodies

### CODE REVIEW

**Comment by troian** [comment]
@boz @Ropes pay special review attention to refactorings of `if` statements

File: provider/bidengine/service.go [hunk 1]
```diff
@@ -129,29 +129,26 @@ loop:
...
+			if evt, ok := ev.(mtypes.EventOrderCreated); ok {
 				// new order
-
-				key := mquery.OrderPath(ev.ID)
+				key := mquery.OrderPath(evt.ID)
 
 				s.session.Log().Info("order detected", "order", key)
 
-				if order := s.orders[key]; order != nil {
+				if ord := s.orders[key]; ord != nil {
```
Comment (boz) [inline provider/bidengine/service.go:138]: did it complain about this name?

File: x/deployment/query/querier.go [hunk 2]
```diff
@@ -33,39 +33,17 @@ func queryDeployments(ctx sdk.Context, path []string, req abci.RequestQuery, kee
 
 	var values Deployments
 	keeper.WithDeployments(ctx, func(deployment types.Deployment) bool {
-		if filters.Owner.Empty() && !isValidState {
+		if (filters.Owner.Empty() && !isValidState) ||
+			(filters.Owner.Empty() && (deployment.State == filters.State)) ||
+			(!isValidState && (deployment.DeploymentID.Owner.Equals(filters.Owner))) ||
+			(deployment.DeploymentID.Owner.Equals(filters.Owner) && deployment.State == filters.State) {
```
Comment (boz) [inline x/deployment/query/querier.go:39]: Put these in a `filter.Accept(types.Deployment) bool` method or something.

File: provider/bidengine/service.go [hunk 3]
```diff
@@ -129,29 +129,26 @@ loop:
 			break loop
 
 		case ev := <-s.sub.Events():
-			switch ev := ev.(type) {
-			case mtypes.EventOrderCreated:
+			if evt, ok := ev.(mtypes.EventOrderCreated); ok {
```
Comment (boz) [inline provider/bidengine/service.go:132]: did it complain about this?

Comment (troian) [reply provider/bidengine/service.go:132]: should not be in here

Comment (troian) [reply provider/bidengine/service.go:132]: It did about switch. or you mean `evt` rename?

Comment (boz) [reply provider/bidengine/service.go:132]: yeah, the `evt` name.

**Comment by troian** [comment]
I would still live it on but suppress for specific cases. When more cases added suppress can be removed

```go
switch bla { // nolint: gocritic

}
```

**Comment by boz** [comment]
we can clean up those conditionals later if you want to get this in.

**Comment by troian** [comment]
Thats actually makes sense. Updated

**Pull request merged**